Added preview-as anonymous/free/paid to post preview modal

kevinansfield opened the pull request:
closes https://linear.app/ghost/issue/PLG-352
closes https://linear.app/ghost/issue/PLG-341

- added preview-as support to previews endpoint
  - updated `/p/` endpoint to accept an optional `?member_status` query parameter with three allowed values: `anonymous/free/paid`
  - when a `member_status` query param is provided:
    - `frame.apiType` is set to `content` which triggers the content-gating paths of our output serializer's post mapper
    - where needed, `frame.original.content.member` is set to a minimal member object with a `status` property to allow content-gating NQL queries to match
    - the combination of both changes our API output to match what the Content API does when rendering posts with a particular logged-in member status
  - if no `member_status` query param is provided the existing behaviour is preserved of rendering as we do for Admin API requests, showing content for all segments simultaneously
- improved preview modal design
  - Added view options for anonymous visitors, free members, and paid members on web.
  - Updated preview toggles to accommodate for web and mobile previews in email.
  - Removed social preview as these are duplicating the functionality in the PSM. They are also not strictly previews, but rather settings.
  - Moved all preview actions (send test email, copy preview URL, open in new tab) to the top bar of the preview to make them more explicit and discoverable.
  - Made the newsletter dropdown and the subject edit field more explicit/discoverable
  - Aligned the publish and update modals header style with the preview modal header style
  - Added a new tooltip component, because the data-tooltip attribute doesn't allow for styling inner content such as shortcuts.
  - Aligned tooltip styles with the Koenig shortcut tooltip
  - Updated post-preview dark mode styles
  - Moved share actions into dropdown
    - Combined copy and open in new tab actions into a dropdown to make it less cluttered and ambiguous

sanne-san committed:
Improved post preview design

ref https://linear.app/ghost/issue/PLG-343/improve-the-preview-modal-design

- Added view options for anonymous visitors, free members, and paid members on web.
- Updated preview toggles to accommodate for web and mobile previews in email.
- Removed social preview as these are duplicating the functionality in the PSM. They are also not strictly previews, but rather settings.
- Moved all preview actions (send test email, copy preview URL, open in new tab) to the top bar of the preview to make them more explicit and discoverable.
- Made the newsletter dropdown and the subject edit field more explicit/discoverable
- Aligned the publish and update modals header style with the preview modal header style
- Added a new tooltip component, because the data-tooltip attribute doesn't allow for styling inner content such as shortcuts.
- Aligned tooltip styles with the Koenig shortcut tooltip
- Updated post-preview dark mode styles
- Moved share actions into dropdown
  - Combined copy and open in new tab actions into a dropdown to make it less cluttered and ambiguous

kevinansfield committed:
Added preview-as anonymous/free/paid to previews endpoint

ref https://linear.app/ghost/issue/PLG-352

- updated `/p/` endpoint to accept an optional `?memberStatus` query parameter with three allowed values: `anonymous/free/paid`
- when a `memberStatus` query param is provided:
  - `frame.apiType` is set to `content` which triggers the content-gating paths of our output serializer's post mapper
  - where needed, `frame.original.content.member` is set to a minimal member object with a `status` property to allow content-gating NQL queries to match
  - the combination of both changes our API output to match what the Content API does when rendering posts with a particular logged-in member status
- if no `memberStatus` query param is provided the existing behaviour is preserved of rendering as we do for Admin API requests, showing content for all segments simultaneously

kevinansfield committed:
fixed template lint issues

kevinansfield committed:
added email address to "From" dropdown

kevinansfield committed:
wired up copy-draft-link button with success state

kevinansfield committed:
fixed test for single/multiple newsletter behaviour, added from address to dropdown

kevinansfield committed:
wired up preview-as selection

sanne-san committed:
Merge branch 'main' into preview-as-minimal

kevinansfield committed:
fixed share dropdown opening test email dropdown

sanne-san committed:
Replaced external icon with copy icon in post preview

Ref https://linear.app/ghost/issue/PLG-348/wire-up-the-post-preview-modal-functionalities

kevinansfield committed:
wired up email desktop/mobile switcher

kevinansfield committed:
added email mobile size change

kevinansfield committed:
fixed remembering format/size when switching between preview/publish

sanne-san committed:
Added styles for mobile email preview

Ref https://linear.app/ghost/issue/PLG-348/wire-up-the-post-preview-modal-functionalities

sanne-san committed:
Hooked up help text for sending test emails

Ref https://linear.app/ghost/issue/PLG-348/wire-up-the-post-preview-modal-functionalities

kevinansfield committed:
fixed email subject changes from preview modal

kevinansfield committed:
fixed preview segment select test

Ghost-Slimer commented:
Deployed to **staging** with ID: `3306`

<sub><sup><a href="https://www.notion.so/ghost/595836ed586a4d3cb93440b6e3a68c52">How does this work?</a></sup></sub>

sanne-san committed:
Improved email preview for mobile

Ref https://linear.app/ghost/issue/PLG-348/wire-up-the-post-preview-modal-functionalities
- Changed subject input to be read-only text on mobile preview
- Removed email address from mobile preview
- Removed accidental fill on "view site" external icon in main nav

Ghost-Slimer commented:
Deployed to **staging** with ID: `3306`

<sub><sup><a href="https://www.notion.so/ghost/595836ed586a4d3cb93440b6e3a68c52">How does this work?</a></sup></sub>

kevinansfield committed:
switched from ?memberStatus to ?member_status

no issue

- we want to be consistent with snake_case query param naming

kevinansfield committed:
Merge branch 'main' into preview-as-minimal

kevinansfield committed:
Merge branch 'main' into preview-as-minimal

Ghost-Slimer commented:
Deployed to **staging** with ID: `3306`

<sub><sup><a href="https://www.notion.so/ghost/595836ed586a4d3cb93440b6e3a68c52">How does this work?</a></sup></sub>

Ghost-Slimer commented:
Deployed to **staging** with ID: `3306`

<sub><sup><a href="https://www.notion.so/ghost/595836ed586a4d3cb93440b6e3a68c52">How does this work?</a></sup></sub>

kevinansfield committed:
Fixed Admin test helpers not correctly updating mocked db settings

no issue

- if helpers were called after some db settings had already been set then they were creating new settings rather than updating existing ones meaning the helper call didn't have the expected effect when the test is run
- switching to the correct `.findBy()` API fixes the conditional so an existing setting is updated

kevinansfield committed:
removed paid member preview options when paid members is not enabled

kevinansfield committed:
fix lint

kevinansfield committed:
fix lint

kevinansfield committed:
Merge branch 'main' into preview-as-minimal

Ghost-Slimer commented:
Deployed to **staging** with ID: `3306`

<sub><sup><a href="https://www.notion.so/ghost/595836ed586a4d3cb93440b6e3a68c52">How does this work?</a></sup></sub>
